### PR TITLE
Problem: rocksdb is out of date

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- upgrade to v8.11.3

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Upgraded RocksDB to version `v8.11.3` for improved store functionality.
- **Refactor**
	- Updated dependencies and configurations to support the new RocksDB version.
	- Modified build and installation steps for macOS and Linux to accommodate RocksDB changes.
- **Chores**
	- Updated the linting process to exclude specific build tags, enhancing code quality checks.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Improvements
+
+* (store) [#]() Upgrade rocksdb to `v9.0.0`.
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
* (store) [#1378](https://github.com/crypto-org-chain/cronos/pull/1378) Upgrade rocksdb to `v9.0.0`.
```

File: go.mod [hunk 2]
```diff
@@ -244,7 +244,6 @@ replace (
 	// TODO Remove it: https://github.com/cosmos/cosmos-sdk/issues/10409
 	github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.9.0
 	github.com/jhump/protoreflect => github.com/jhump/protoreflect v1.9.0
-	github.com/linxGnu/grocksdb => github.com/linxGnu/grocksdb v1.8.0
```
Comment (mmsqe) [inline go.mod:247]: https://github.com/linxGnu/grocksdb/compare/v1.8.0...v1.8.14
https://github.com/facebook/rocksdb/compare/v8.1.1...v8.11.3

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Improvements
+
+* (store) [#1378](https://github.com/crypto-org-chain/cronos/pull/1378) Upgrade rocksdb to `v9.0.0`.
```
Comment (mmsqe) [inline CHANGELOG.md:7]: you mean v8.11.3 right?

Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
* (store) [#1378](https://github.com/crypto-org-chain/cronos/pull/1378) Upgrade rocksdb to `v8.11.3`.
```

Comment (yihuang) [reply CHANGELOG.md]: yes

**Review by mmsqe — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Improvements
+
+* (store) [#1378](https://github.com/crypto-org-chain/cronos/pull/1378) Upgrade rocksdb to `v8.11.3`.
```
[the remark this answers is not in the document]
Comment (yihuang) [reply CHANGELOG.md:7]: done, thanks

**Comment by mmsqe** [comment]
should we align memiavl as well? but do we do any db migrate to avoid the invalid nonce err

**Comment by yihuang** [comment]
> should we align memiavl as well?

they only depends on grocksdb indirectly, no necessary to update manually. 

> but do we do any db migrate to avoid the invalid nonce err

this is needed.

**Comment by mmsqe** [comment]
> 
> this is needed.

I got invalid nonce bump from 8.1.1 to 8.3.2

**Comment by yihuang** [comment]
> > this is needed.
> 
> I got invalid nonce bump from 8.1.1 to 8.3.2

yeah, still investigating.

**Pull request merged**